src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -0,0 +1,445 @@
+
+        internal static class FileStream
+        {
+            internal const string TypeName = "System.IO.FileStream";
+
+            internal static readonly Type Type = GetTypeFromEither(
+                contractName: $"${TypeName}, ${System_IO_FileSystem_Name}",
+                desktopName: TypeName);
+
+            internal static readonly PropertyInfo Name = Type
+                .GetTypeInfo()
+                .GetDeclaredProperty(nameof(Name));
+
+            private static ConstructorInfo s_Ctor_String_FileMode_FileAccess_FileShare = Type
+                .GetTypeInfo()
+                .GetDeclaredConstructor(paramTypes: new[] { typeof(string), FileMode.Type, FileAccess.Type, FileShare.Type });
+
+            private static ConstructorInfo s_Ctor_String_FileMode_FileAccess = Type
+                .GetTypeInfo()
+                .GetDeclaredConstructor(paramTypes: new[] { typeof(string), FileMode.Type, FileAccess.Type });
+
+            private static ConstructorInfo s_Ctor_String_FileMode = Type
+                .GetTypeInfo()
+                .GetDeclaredConstructor(paramTypes: new[] { typeof(string), FileMode.Type });
+
+            private static ConstructorInfo s_Ctor_String_FileMode_FileAccess_FileShare_Int32_FileOptions = Type
+                .GetTypeInfo()
+                .GetDeclaredConstructor(paramTypes: new[] { typeof(string), FileMode.Type, FileAccess.Type, FileShare.Type, typeof(int), FileOptions.Type });
+
+            public static Stream Create(string path, object mode)
+            {
+                try
+                {
+                    return (Stream)s_Ctor_String_FileMode.Invoke(new[] { path, mode });
+                }
+                catch (TargetInvocationException e)
+                {
+                    throw e.InnerException;
+                }
+            }
+
+            public static Stream Create(string path, object mode, object access)
+            {
+                try
+                {
+                    return (Stream)s_Ctor_String_FileMode_FileAccess.Invoke(new[] { path, mode, access });
+                }
+                catch (TargetInvocationException e)
+                {
+                    throw e.InnerException;
+                }
+            }
+
+            public static Stream Create(string path, object mode, object access, object share)
+            {
+                try
+                {
+                    return (Stream)s_Ctor_String_FileMode_FileAccess_FileShare.Invoke(new[] { path, mode, access, share });
+                }
+                catch (TargetInvocationException e)
+                {
+                    throw e.InnerException;
+                }
+            }
+
+            public static Stream CreateEx(string path, object mode, object access, object share)
+            {
+                try
+                {
+                    return Create(path, mode, access, share);
+                }
+                catch (TargetInvocationException e)
```

Comment:
You'll want to catch InvalidOperationException here as well if these code paths run on Win8. On some platforms a(Win8), the types exist underneath, but are not callable. For example, setting the CurrentUICulture is one of those.